### DISCUSSION TITLE
How to just display text without enter a reactive mode?

### DISCUSSION THREAD

**Discussion by ngdangtu-vn** [Im-Beast/deno_tui#44] [Q&A] [answered]
TUI is powerful, but not every part in my cli need tui. For instance, I have a "help" page and I just need to print out then done. Do we have any option to just print out then exit (`Deno.exit(0)`)?

I think I saw something like `TuiOptions.stdin.isTerminal()` but I don't know if it is actually what I want :?

**Comment by Im-Beast** [comment, marked as answer]
You could check out [Nice](https://github.com/Im-Beast/nice/tree/main), another module of mine that sounds exactly what you want. To see how to use it, please go to the examples in the repo.

Keep in mind however that `Nice` is still work in progress, while most things already work, there might be some changes in the future. Any sort of feedback regarding it would be fantastic as well!

**Reply by ngdangtu-vn** [reply]
I would hope a solution come from the same module so I don't have to install more. But it make sense to have a different mod as well since this is not the duty of tui lib. Good thing it come from the same father? At least I could look forward to consitency in code interface right?

**Reply by Im-Beast** [reply]
Nice is a separate package, and you don't need `Tui` to work with it.
Third version of `Tui`, that I'm slowly working on will depend on `Nice` and use it behind the scenes. I'll try to work on it a bit more in the next 2 weeks to hopefully get something to show off :). The API will be different, but hopefully for the better. Currently code consists of lots of unnecessary boilerplate and it's kind of hard to see what lies where, that's what will be improved in `Tui v3`.

**Reply by ngdangtu-vn** [reply]
I don't mind if this feature got separate package since it will be more convenience different purposes. I just think about the consistency in code between the 2 lib. It good be great if they share the same style.

I just started to use your lib today so I cannot say much. But in my exp, the way to add component to tui kinda caught me off guard a bit. Like usually, when you create a new instance from a class, you would expect it to be used somewhere right? It kinda surprise me at first when I only create a new instance and leave it... there. I wonder how it works behind the scence. How does tui instance know when a new component created. Maybe sth like this could make code more readable :?

```js
import {TUI} from 'tui/mod.ts'
import {Button} from 'tui/components/button.ts'

const tui = TUI.getInstance({/* ... */})

tui.add(Button, 'button content', {/* opt */})

tui.run() // for reactive page (interact ui)
tui.print() // for static page
```

**Reply by Im-Beast** [reply]
`Tui 3` will use JSX, so new API will look (somewhat) like this:

```tsx
import Button from ...;
import Text from ...;
import { render } from ...;

interface State {
  counter: number;
}

function App(state: State) {
  return <>
    <Button onClick={() => (state.counter += 1)}>Hello!</Button>
    <Text>Current count: {state.counter}</Text>
  </>;
}

const state = render(<App counter={0} />); // start rendering
state.stop(); // stop rendering
```
